homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,491 @@
+) -> None:
+    """Set up entry."""
+    _LOGGER.debug("Setting up switches")
+    fritzbox_tools: FritzBoxTools = hass.data[DOMAIN][entry.entry_id]
+
+    _LOGGER.debug("Fritzbox services: %s", fritzbox_tools.connection.services)
+
+    entities = await get_entities_list(fritzbox_tools, entry.title)
+    async_add_entities(entities)
+
+
+class FritzBoxPortSwitch(FritzBoxBaseSwitch, SwitchEntity):
+    """Defines a FRITZ!Box Tools PortForward switch."""
+
+    def __init__(
+        self,
+        fritzbox_tools: FritzBoxTools,
+        device_friendly_name,
+        port_mapping,
+        idx,
+        connection_type,
+    ):
+        """Init Fritzbox port switch."""
+        self.fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self._attributes = {}
+        self.connection_type = connection_type
+        self.port_mapping: dict = port_mapping  # dict in the format as it comes from fritzconnection. eg: {'NewRemoteHost': '0.0.0.0', 'NewExternalPort': 22, 'NewProtocol': 'TCP', 'NewInternalPort': 22, 'NewInternalClient': '192.168.178.31', 'NewEnabled': True, 'NewPortMappingDescription': 'Beast SSH ', 'NewLeaseDuration': 0}
+        self._idx = idx  # needed for update routine
+
+        switch_info = SwitchInfo(
+            description=f'Port forward {port_mapping["NewPortMappingDescription"]}',
+            friendly_name=device_friendly_name,
+            icon="mdi:check-network",
+            type=SWITCH_TYPE_PORTFORWARD,
+            callback_update=self._async_fetch_update,
+            callback_switch=self._async_handle_port_switch_on_off,
+        )
+        super().__init__(fritzbox_tools, switch_info)
+
+    async def _async_fetch_update(self):
+        """Fetch updates."""
+
+        self.port_mapping = await service_call_action(
+            self.fritzbox_tools,
+            self.connection_type,
+            "1",
+            "GetGenericPortMappingEntry",
+            NewPortMappingIndex=self._idx,
+        )
+        _LOGGER.debug(
+            "Specific %s response: %s", SWITCH_TYPE_PORTFORWARD, self.port_mapping
+        )
+        self._attr_is_on = self.port_mapping["NewEnabled"] is True
+        self._is_available = True
+
+        attributes_dict = {
+            "NewInternalClient": "internalIP",
+            "NewInternalPort": "internalPort",
+            "NewExternalPort": "externalPort",
+            "NewProtocol": "protocol",
+            "NewPortMappingDescription": "description",
+        }
+
+        for key in attributes_dict:
+            self._attributes[attributes_dict[key]] = self.port_mapping[key]
+
+    async def _async_handle_port_switch_on_off(self, turn_on: bool) -> bool:
+
+        self.port_mapping["NewEnabled"] = "1" if turn_on else "0"
+
+        resp = await service_call_action(
+            self.fritzbox_tools,
+            self.connection_type,
+            None,
+            "AddPortMapping",
+            **self.port_mapping,
+        )
+
+        if not resp:
```

Comment:
My response is always `{}` which evaluates this expression to True.

Reply:
```suggestion
        if resp is None:
```

Reply:
Ok, will change to `if resp is None:`

Reply:
Fixed